source/request-response/weak_ref.c

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/mqtt/private/request-response/weak_ref.h>
+
+#include <aws/common/ref_count.h>
+
+struct aws_weak_ref {
+    struct aws_allocator *allocator;
+    struct aws_ref_count refcount;
+    void *referenced;
+};
+
+static void s_destroy_weak_ref(void *value) {
```

Comment:
nitpick/trivial: structure of [subject][verb][qualifiers] in `destroy` being at the end of most of these calls. I found we use a lot of `s_destroy_[subject]` in our mqtt5 implementation so maybe this is an alternate pattern we use so we can keep as is?

Reply:
The names of internal static functions aren't really critical